---
title: "Quantitative Traits"
engine: knitr
format: 
  html: 
    toc: true
    toc-depth: 3
    toc-title: Contents
    self-contained: true
    css: ../custom.css
    link-external-newwindow: true
webr: 
  packages: ["ggplot2", "dplyr", "tibble"]
filters: 
  - webr
bibliography: ../Genetics_Discussion.bib
csl: ../evolution.csl
code-annotations: hover
---


```{r}
#| echo: false
#| message: false
#| warning: false

library(ggplot2)
library(dplyr)
library(tibble)
```


## Preliminaries

If you are already familiar with the structure of these exercises, read the [Introduction](https://biosc-2200.github.io/Introduction/) first.

::: {.callout-note}

Reminder: Save your work regularly.

:::

::: {.callout-important}

If you are using a Mac, we recommend that you use either Chrome or Firefox to complete these exercises. Some of the default settings in Safari prevent these exercises from running.

:::


### Contact information

If you have questions about these exercises, please contact Dr. Kevin Middleton (middletonk@missouri.edu) or drop by Tucker 224.



## Learning Objectives {#sec-objectives}

The learning objectives for this exercise are:

- Explain how polygenic traits differ from Mendelian traits
- Explain how traits with continuous (also called quantitative) phenotypic measures result from the combined effects of many different genes
- Describe how many genes can each contribute a small amount to a phenotype
- Explain what quantitative trait loci (QTL) are and how QTL are discovered
- Explain how the contributions of many genes of small effect can be associated with a disease or condition


## Contrasting Mendelian traits and polygenic traits

Dominant/recessive to just thinking about alternate alleles (major vs. minor)


## What are quantitative traits?


## Counting the ways: Binomial Coefficent



## Continuous traits from combinations of many Mendelian traits

```{r}
#| echo: false
#| fig-align: center
#| fig-height: 4

source("genotypes_plot.R")
genotypes_plot(n_genes = 2)

```


```{webr-r}
#| context: setup

library(ggplot2)
library(dplyr)
library(tibble)

source("https://raw.githubusercontent.com/BIOSC-2200/BIOSC-2200.github.io/main/QT/genotypes_plot.R")
```



```{webr-r}
#| fig-height: 4
genotypes_plot(n_genes = 3)
```


### Combinations of alleles are binomial


### Large numbers of small additions and subtractions are normal

Make some assumptions:

- Additivity can mean adding negative numbers
- All genes have roughly equal effect
- Gene do not interact with one another


## Case study: Human height


```{r}
#| echo: false

NH <- read.csv("NHANES.csv")

NHa <- NH |> 
  filter(Age > 20)
```

The National Health and Nutrition Examination Survey (["NHANES"](https://wwwn.cdc.gov/nchs/nhanes/Default.aspx)) began in the early 1960's and continues to this day. The goal is to assess the health and nutrition status of a broad cross-section of the population. As part of this study, routine measurments of body form such as height (in cm) and body mass (in kg) are recorded for each participant.

The 2017-2020 NHANES survey has data for `r nrow(NH)` individuals. @fig-htall shows...

```{r}
#| label: fig-htall
#| echo: false
#| fig-height: 4
#| fig-cap: FIXME. Data from [NHANES 2017-2020](https://wwwn.cdc.gov/nchs/nhanes/Default.aspx.
#| cap-location: bottom

ggplot(NH, aes(x = Age, y = Height, color = Genotype)) +
  geom_point(size = 1, show.legend = FALSE, alpha = 0.25,
             position = position_jitter(width = 0.5, seed = 3847293)) +
  scale_color_manual(values = c("goldenrod", "firebrick")) +
  scale_x_continuous(breaks = seq(0, 80, by = 5)) +
  facet_grid(Genotype ~ .) +
  labs(x = "Age (y)", y = "Height (cm)") +
  mytheme
```



```{r}
#| label: fig-ht
#| echo: false
#| fig-height: 4
#| fig-cap: FIXME. Data from [NHANES 2017-2020](https://wwwn.cdc.gov/nchs/nhanes/Default.aspx.
#| cap-location: bottom

NHa_means <- NHa |> 
  group_by(Genotype) |> 
  summarize(across(.cols = everything(),
                   list(mean = mean, sd = sd)))

ggplot(NHa, aes(Height, fill = Genotype)) +
  geom_histogram(bins = 30, show.legend = FALSE) +
  scale_fill_manual(values = c("goldenrod", "firebrick")) +
  facet_grid(Genotype ~ .) +
  labs(x = "Height (cm)", y = "Count") +
  mytheme

```

## Generating a normal distribution from combinations of alleles

```{webr-r}
simulate_heights(n_genes = 25)
```


## Describing distributions

### Using distributions


## Associating QTLs with genetic variants

Intro SNP


### Shapiro pigeon example (dominant trait)

![FIXME Image from [@Shapiro2013-hx].](../images/pigeon1.png){width=100% fig-align="left" fig-alt="" #fig-pigeon1}

![FIXME Image from [@Shapiro2013-hx].](../images/pigeon2.png){width=100% fig-align="left" fig-alt="" #fig-pigeon2}


## QTL for Human Height

![FIXME Image from [@Visscher2012-ph].](../images/gr1_lrg.jpg){width=75% fig-align="center" fig-alt="" #fig-gwas1}

- Best understood quantitative trait in humans
- Yet still 700 genes


![FIXME Image from [@Visscher2017-il].](../images/gr2_lrg.jpg){width=100% fig-align="center" fig-alt="" #fig-gwas1}


Yengo et al. [-@Yengo2018-gk]:

- ~700,000 individuals
- 3290 ("near-independent") SNPs explain ~25% of the variation in human height among Europeans
- Estimated to be ~700 explaining ~16% of variation in 2010 [@Lango_Allen2010-iu]



## Case Study: Threshold traits

### Schizophrenia (~200 genes)

Why family history is one of the most important diagnostic tools in medicine


## References

::: {#refs}
:::
