---
title: "Quantitative Traits"
engine: knitr
format: 
  html: 
    toc: true
    toc-depth: 3
    toc-title: Contents
    self-contained: true
    css: ../custom.css
    link-external-newwindow: true
webr: 
  packages: ["ggplot2", "dplyr", "tibble"]
filters: 
  - webr
bibliography: ../Genetics_Discussion.bib
csl: ../evolution.csl
code-annotations: hover
---


```{r}
#| echo: false
#| message: false
#| warning: false

library(ggplot2)
library(dplyr)
library(tibble)

source("genotypes_plot.R")

```


```{webr-r}
#| context: setup

library(ggplot2)
library(dplyr)
library(tibble)

source("https://raw.githubusercontent.com/BIOSC-2200/BIOSC-2200.github.io/main/QT/genotypes_plot.R")

download.file(
  "https://raw.githubusercontent.com/BIOSC-2200/BIOSC-2200.github.io/main/QT/NHANES/NHANES.csv",
  "NHANES.csv",
  quiet = TRUE)

```


## Preliminaries

If you are already familiar with the structure of these exercises, read the [Introduction](https://biosc-2200.github.io/Introduction/) first.

::: {.callout-note}

Reminder: Save your work regularly.

:::

::: {.callout-important}

If you are using a Mac, we recommend that you use either Chrome or Firefox to complete these exercises. Some of the default settings in Safari prevent these exercises from running.

:::


### Contact information

If you have questions about these exercises, please contact Dr. Kevin Middleton (middletonk@missouri.edu) or drop by Tucker 224.


## Learning objectives

The learning objectives for this exercise are:

- Explain how polygenic traits differ from Mendelian traits
- Explain how quantitative traits with continuous-valued phenotypic measures result from the combined effects of many different genes
- Describe how many genes can each contribute a small amount to a measurable phenotype
- Explain what quantitative trait loci (QTL) are and how QTL are identified
- Explain how the contributions of many genes of small effect can be associated with a disease or condition


## Contrasting Mendelian traits and polygenic traits

The first phenotypes that you learned about as well as those described in the first in this series of exercises (<a href="./TGI" target="_blank">Transmission of Genetic Information</a>) were Mendelian traits. In Mendelian traits, a single gene is responsible for a single trait. In this context, you also learned about dominant and recessive alleles (and their variations), which lead to different observable phenotypes.

This set of exercises focuses on phenotypes that are determined by multiple genes: **polygenic traits**. Polygenic traits often (but not always) can be measured on a numeric scale: *quantitative traits*. The two main types of quantitative traits are:

- *Meristic traits*: Traits that take on integer values such as the number of peas in a pod. 3, 4, and 5 are all possible values, but a pod can't have 3.5 peas.
- *Continuous-valued traits*: Traits that can take on any number on the number line. For example lengths and weights can have any value depending on the scale (e.g., grams, pounds, etc.)

FIXME

Dominant/recessive to just thinking about alternate alleles (major vs. minor)


### Alleles associated with quantitative traits


## Counting the ways that alleles can combine

![The process of gamete formation leads to predictable genotypic and phenotypic proportions in the F~2~ generation. Image modified from Klug et al. [-@Klug2019-cs]](../images/punnett.png){fig-alt="" #fig-punnett}

The 1:2:1 genotypic ratio and 3:1 phenotypic ratio in the heterozygote cross example for a dominant Mendelian trait (@fig-punnett) represent theoretical probabilities for the distributions of genotypes and phenotypes. Higher level crosses, with two and three genes can be done manually, but very quickly you will find that keeping track of all the combinations becomes very challenging (@fig-trihybrid).

![Punnett square for a trihybrid cross in peas.](https://pbs.twimg.com/media/CvuLNpHXgAQpSwh.jpg){fig-alt="" #fig-trihybrid}

Punnett squares with more than 3 genes get very difficult (4 genes requires 256 combinations, and 5 needs 1,024 combinations). Fortunately, rather than keeping track of all these combinations manually, we can just calculate them directly using a little bit of math.

Before we think about combining alleles, let's just consider something simpler: flipping coins. Imagine that you flip a coin twice. Each flip can result in either heads or tails. With two coin flips there are four possibilities:

1. Heads, Heads
2. Heads, Tails
3. Tails, Heads
4. Tails, Tails

To get two heads or two tails, both coin flips have to be the same. But to get one head and one tail, there are two possibilities:

- Heads, Tails
- Tails, Heads

Because the sequence doesn't matter, both result in one of each (head and tail). If we then think about adding up the possible sets of results, we have three possibilities:

1. 2 Heads (1 way)
2. 1 Head, 1 Tail (2 ways)
3. 2 Tails (1 way)

If you look back up at the monohybrid cross above, you will find that there is 1 DD, 2 Dd, and 1 dd. This 1:2:1 genotypic ratio is the same as for our coin flipping example.

All the different ways that you can arrive at a count of heads from a set of coin tosses is represented by a number called the *Binomial Coefficient*. The equation for the binomial coefficient for the number of heads from a set of coin flips is:

$$\frac{Flips!}{Heads!(Flips - Heads)!}$$

Those exclamation points (!) are the factorial function (3! = 3 x 2 x 1 = 6). If we plug in the numbers for 1 Head from 2 Flips:

$$\frac{2!}{1!(2 - 1)!}$$

which reduces to:

$$\frac{2 \times 1}{1(1)}$$

which is just $\frac{2}{1}$ or 2. So there are 2 ways to get 1 head from 2 coin flips, just like we figured out manually.

Fortunately, we don't have to do all these factorials by hand when there are more genotypes. We can calculate the binomial coefficient directly using the `choose()` function. Run the code below to confirm that there are 1, 2 and 1 ways to get different genotypes:

```{webr-r}
choose(n = 2, k = 0) # dd
choose(n = 2, k = 1) # Dd or dD
choose(n = 2, k = 2) # DD
```

Feel free to change the values for `n` (the number of "trials") and `k` (the number of "successes") and try out some additional combinations.^[The binomial coefficient calculation is also used to calculate probabilities for lotteries (the Powerball odds of winning is 1 in `choose(69, 5) * choose(26, 1)`) and poker hands (the odds of a royal flush in 5-card draw is 4 in `choose(52, 5)` or equivalently one in `choose(52, 5) / 4`). In fact, much of the basics of probability were originally developed in the 17th and 18th centuries to try to understand various forms of gambling.]


### Replace coin flips with independent alleles



A and T FIXME

@fig-genesplot shows...FIXME

```{r}
#| label: fig-genesplot
#| echo: false
#| fig-height: 4
#| fig-cap: FIXME. 
#| cap-location: bottom

genotypes_plot(n_genes = 1)
```

The figure below calculates all the combinations for a trihybrid cross, like the example of corn above. Notice how the grid above is 8 by 8, giving 64 possible combinations. The total combinations below is also 64 (but with a lot less ) 

```{webr-r}
#| fig-height: 4
genotypes_plot(n_genes = 2)
```



Combinations of alleles are binomial


## Continuous phenotypes result from combinations of many alleles


### Large numbers of small additions and subtractions are normal

Make some assumptions:

- Additivity can mean adding negative numbers
- All genes have roughly equal effect
- Gene do not interact with one another


## Case study: Human height


```{r}
#| echo: false

NH <- read.csv("NHANES/NHANES.csv")

NHa <- NH |> 
  filter(Age > 20)
```

The National Health and Nutrition Examination Survey (["NHANES"](https://wwwn.cdc.gov/nchs/nhanes/Default.aspx)) began in the early 1960's and continues to this day. The goal is to assess the health and nutrition status of a broad cross-section of the population. As part of this study, routine measurements of body size such as height (in cm) are recorded for each participant.

The 2017-2020 NHANES survey has data for `r format(nrow(NH), big.mark = ",")` individuals.

@fig-htall shows...

```{r}
#| label: fig-htall
#| echo: false
#| fig-height: 4
#| fig-cap: FIXME. Data from [NHANES 2017-2020](https://wwwn.cdc.gov/nchs/nhanes/Default.aspx).
#| cap-location: bottom

ggplot(NH, aes(x = Age, y = Height, color = Genotype)) +
  geom_point(size = 1, show.legend = FALSE, alpha = 0.25,
             position = position_jitter(width = 0.5, seed = 3847293)) +
  scale_color_manual(values = c("goldenrod", "firebrick")) +
  scale_x_continuous(breaks = seq(0, 80, by = 5)) +
  facet_grid(Genotype ~ .) +
  labs(x = "Age (y)", y = "Height (cm)") +
  mytheme
```



```{r}
#| label: fig-ht
#| echo: false
#| fig-height: 4
#| fig-cap: FIXME. Data from [NHANES 2017-2020](https://wwwn.cdc.gov/nchs/nhanes/Default.aspx).
#| cap-location: bottom

NHa_means <- NHa |> 
  group_by(Genotype) |> 
  summarize(across(.cols = everything(),
                   list(mean = mean, sd = sd)))

ggplot(NHa, aes(Height, fill = Genotype)) +
  geom_histogram(bins = 30, show.legend = FALSE) +
  scale_fill_manual(values = c("goldenrod", "firebrick")) +
  facet_grid(Genotype ~ .) +
  labs(x = "Height (cm)", y = "Count") +
  mytheme

```

## Generating a normal distribution from combinations of alleles

@fig-gen_normal FIXME

```{r}
#| label: fig-gen_normal
#| echo: false
#| fig-height: 4
#| fig-cap: FIXME.
#| cap-location: bottom

NH_XX <- read.csv("NHANES/NHANES.csv") |> 
  filter(Genotype == "XX" & Age > 20)
simulate_heights(n_genes = 5, XX = NH_XX)
```

Try increasing numbers of genes (e.g., 10, 20, 50, 100, 300). 

```{webr-r}
simulate_heights(n_genes = 10)
```

> As the number of genes increases, how do the distributions of actual heights and simulated heights compare to one another? How the amount of phenotypic variation attributable to each allele change as the number of genes controbuting to height increases?

<textarea rows="3" cols="75">

</textarea>


## Describing distributions

### Using distributions


## Associating QTLs with genetic variants

Intro SNP


### Shapiro pigeon example (dominant trait)

![FIXME Image from [@Shapiro2013-hx].](../images/pigeon1.png){width=100% fig-align="left" fig-alt="" #fig-pigeon1}

![FIXME Image from [@Shapiro2013-hx].](../images/pigeon2.png){width=100% fig-align="left" fig-alt="" #fig-pigeon2}


## QTL for Human Height

![FIXME Image from [@Visscher2012-ph].](../images/gr1_lrg.jpg){width=75% fig-align="center" fig-alt="" #fig-gwas1}

- Best understood quantitative trait in humans
- Yet still 700 genes


![FIXME Image from [@Visscher2017-il].](../images/gr2_lrg.jpg){width=100% fig-align="center" fig-alt="" #fig-gwas1}


Yengo et al. [-@Yengo2018-gk]:

- ~700,000 individuals
- 3290 ("near-independent") SNPs explain ~25% of the variation in human height among Europeans
- Estimated to be ~700 explaining ~16% of variation in 2010 [@Lango_Allen2010-iu]



## Case Study: Threshold traits

### Schizophrenia (~200 genes)

Why family history is one of the most important diagnostic tools in medicine


## References

::: {#refs}
:::
